srt.go

```diff
@@ -111,7 +113,88 @@ func ReadFromSRT(i io.Reader) (o *Subtitles, err error) {
 			o.Items = append(o.Items, s)
 		} else {
 			// Add text
-			s.Lines = append(s.Lines, Line{Items: []LineItem{{Text: strings.TrimSpace(line)}}})
+			if l := parseTextSrt(line); len(l.Items) > 0 {
+				s.Lines = append(s.Lines, l)
+			}
+			// s.Lines = append(s.Lines, Line{Items: []LineItem{{Text: strings.TrimSpace(line)}}})
+		}
+	}
+	return
+}
+
+func parseTextSrt(i string) (o Line) {
+	// Create tokenizer
+	tr := html.NewTokenizer(strings.NewReader(i))
+
+	// Loop
+	type Styles struct {
+		bold      bool
+		italic    bool
+		underline bool
+		color     *string
+	}
+	styles := Styles{}
+	for {
+		// Get next tag
+		t := tr.Next()
+
+		// Process error
+		if err := tr.Err(); err != nil {
+			break
+		}
+
+		// Get current token
+		token := tr.Token()
+
+		switch t {
+		case html.EndTagToken:
+			// Parse italic/bold/underline
+			switch token.Data {
+			case "b":
+				styles.bold = false
+			case "i":
+				styles.italic = false
+			case "u":
+				styles.underline = false
+			case "font":
+				styles.color = nil
+			}
+		case html.StartTagToken:
+			// Parse italic/bold/underline
+			switch token.Data {
+			case "b":
+				styles.bold = true
+			case "i":
+				styles.italic = true
+			case "u":
+				styles.underline = true
+			case "font":
+				color, _ := getAttribute(&token, "color")
+				if color != "" {
+					styles.color = &color
+				}
+			}
+		case html.TextToken:
+			if s := strings.TrimSpace(string(tr.Raw())); s != "" {
+				// Get style attribute
+				var sa *StyleAttributes
+				if styles.bold || styles.italic ||
+					styles.underline || styles.color != nil {
+					sa = &StyleAttributes{
+						TTMLColor:       styles.color,
```

Comment:
Could you create `StyleAttributes`'s `SRTColor`, `SRTBold`, `SRTItalic` and `SRTUnderline`, and store the values inside them instead?